lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -241,9 +259,19 @@ defmodule ExUnit.Runner do
       end
 
     EM.test_finished(config.manager, test)
+    handle_result(config, test)
     test
   end
 
+  defp handle_result(config, %{state: {:failed, _}}) do
```

Comment:
Micro-optimization: ignore this logic if the `--max-failures` flag is not given.

```elixir
defp handle_result(%{max_failures: max_failures}, %{state: {:failed, _}}) when is_integer(max_failures) do
```